In [3]:
%%capture
!pip install ipython-sql sqlalchemy
import sqlalchemy
sqlalchemy.create_engine("sqlite:///humankind.sqlite")
%load_ext sql
%sql sqlite:///humankind.sqlite
    

In [4]:
%%sql

with cte as (
select distinct m.customerid, c.category, 
count(c.category) over (partition by m.customerid) as count
from messages m
join customers c 
on m.customerid = c.id
and category  like 'new'
order by count desc)

select avg(count)
from cte

 * sqlite:///humankind.sqlite
Done.


[(10.513842281879194,)]

In [5]:
%%sql

with cte as (select distinct m.customerid, c.category, 
count(c.category) over (partition by m.customerid) as count
from messages m
join customers c 
on m.customerid = c.id
and category  like 'repeat'
order by count desc)

select avg(count)
from cte


 * sqlite:///humankind.sqlite
Done.


[(15.177033492822966,)]

In [6]:
%%sql

select count(*) as total_repeat_leads
from customers_4
where category like 'repeat'
and claimedat not  like ''

 * sqlite:///humankind.sqlite
Done.


[(414,)]

In [7]:
%%sql

select count(*) as first_24_hours
from customers_4
where category like 'repeat'
and claimedat not  like ''
and claimdifference between 0 and 1

 * sqlite:///humankind.sqlite
Done.


[(234,)]

In [8]:
%%sql

select count(*) as days2through5
from customers_4
where category like 'repeat'
and claimedat not  like ''
and claimdifference between 2 and 5

 * sqlite:///humankind.sqlite
Done.


[(21,)]

In [9]:
%%sql

select count(*) as days6through10
from customers_4
where category like 'repeat'
and claimedat not  like ''
and claimdifference between 6 and 10

 * sqlite:///humankind.sqlite
Done.


[(34,)]

In [10]:
%%sql

select count(*) as days7through15
from customers_4
where category like 'repeat'
and claimedat not  like ''
and claimdifference between 7 and 15

 * sqlite:///humankind.sqlite
Done.


[(51,)]

In [11]:
%%sql

select count(*) as Repeats_Beyond15days
from customers_4
where category like 'repeat'
and claimedat not  like ''
and claimdifference  > 15

 * sqlite:///humankind.sqlite
Done.


[(106,)]

In [12]:
%%sql

select count(*) as total_new_leads
from customers_4
where category like 'new'
and claimedat not  like ''
 

 * sqlite:///humankind.sqlite
Done.


[(2384,)]

In [13]:
%%sql

select count(*) as first_24_hours
from customers_4
where category like 'new'
and claimedat not  like ''
and claimdifference between 0 and 1

 * sqlite:///humankind.sqlite
Done.


[(1233,)]

In [14]:
%%sql

select count(*) as days2through5
from customers_4
where category like 'new'
and claimedat not  like ''
and claimdifference between 2 and 5

 * sqlite:///humankind.sqlite
Done.


[(332,)]

In [15]:
%%sql

select count(*) as days6through10
from customers_4
where category like 'new'
and claimedat not  like ''
and claimdifference between 6 and 10

 * sqlite:///humankind.sqlite
Done.


[(231,)]

In [16]:
%%sql

select count(*) as days7through15
from customers_4
where category like 'new'
and claimedat not  like ''
and claimdifference between 7 and 15

 * sqlite:///humankind.sqlite
Done.


[(433,)]

In [17]:
%%sql

select count(*) as NewLeads_Beyond15days
from customers_4
where category like 'new'
and claimedat not  like ''
and claimdifference >15

 * sqlite:///humankind.sqlite
Done.


[(379,)]

In [18]:
%%sql

select count(*) 
from customers_4
where category not like 'lead'
and claimedat  not like ''


 * sqlite:///humankind.sqlite
Done.


[(2798,)]

We have 2798 leads total excluding the category "lead" as well as the records fall under repeat or new that were never claimed. Of the 2798 total leads, 1467 were claimed in the first 24 hours. Of those 1467, 234(8%) converted to a "repeat" meaning they purchased. There were 353 leads claimed 2-5 days after they signed up. Of these 353, 6% converted to a "repeat. There were 265 leads claimed 6-10 days after sign up. 34(13%) converted to a repeat. 484 leads were claimed 7-15 days after signing up. Of these 484, 51(11%) converted. 485 leads were claimed after more than 15 days passed. Of these 485, 22% converted

In [19]:
%%sql

select e.expert,  claimedatdate, count(c.expertid)
from customers_4 c
join experts_1 e on e.expertid = c.expertid
group by e.expert,claimedatdate
order by claimedatdate desc


 * sqlite:///humankind.sqlite
Done.


[('Bob Dole', '9/28/21', 11),
 ('Bob Dole', '9/24/21', 30),
 ('Bob Dole', '9/23/21', 36),
 ('Bob Dole', '9/22/21', 57),
 ('Emma Stone', '9/22/21', 4),
 ('Bob Dole', '9/21/21', 6),
 ('Humankind Admin', '9/17/21', 2),
 ('Bob Dole', '5/9/22', 40),
 ('Bob Dole', '5/6/22', 28),
 ('Humankind Admin', '5/6/22', 1),
 ('Bob Dole', '5/5/22', 37),
 ('Humankind Admin', '5/5/22', 1),
 ('Bob Dole', '5/4/22', 14),
 ('Bob Dole', '5/3/22', 7),
 ('Bob Dole', '5/2/22', 21),
 ('Bob Dole', '4/7/22', 73),
 ('Bob Dole', '4/6/22', 50),
 ('Bob Dole', '4/5/22', 57),
 ('Bob Dole', '4/4/22', 42),
 ('Bob Dole', '4/29/22', 7),
 ('Bob Dole', '4/28/22', 11),
 ('Bob Dole', '4/27/22', 17),
 ('Bob Dole', '4/26/22', 50),
 ('Bob Dole', '4/25/22', 54),
 ('Bob Dole', '4/22/22', 41),
 ('Bob Dole', '4/21/22', 71),
 ('Bob Dole', '4/20/22', 46),
 ('Bob Dole', '4/19/22', 90),
 ('Bob Dole', '4/18/22', 46),
 ('Bob Dole', '4/14/22', 52),
 ('Bob Dole', '4/13/22', 34),
 ('Bob Dole', '4/12/22', 24),
 ('Bob Dole', '4/11/22', 62),
 ('Bob Dole', '4/1/22', 9),
 ('Bob Dole', '3/9/22', 58),
 ('Bob Dole', '3/8/22', 112),
 ('Bob Dole', '3/31/22', 8),
 ('Bob Dole', '3/30/22', 10),
 ('Bob Dole', '3/29/22', 4),
 ('Bob Dole', '3/25/22', 50),
 ('Bob Dole', '3/24/22', 137),
 ('Bob Dole', '3/23/22', 9),
 ('Bob Dole', '3/11/22', 35),
 ('Bob Dole', '3/10/22', 53),
 ('Bob Dole', '3/1/22', 93),
 ('Bob Dole', '2/9/22', 4),
 ('Bob Dole', '2/8/22', 7),
 ('Bob Dole', '2/7/22', 18),
 ('Bob Dole', '2/4/22', 6),
 ('Bob Dole', '2/3/22', 10),
 ('Bob Dole', '2/25/22', 69),
 ('Bob Dole', '2/24/22', 13),
 ('Bob Dole', '2/23/22', 1),
 ('Bob Dole', '2/22/22', 12),
 ('Bob Dole', '2/2/22', 8),
 ('Bob Dole', '2/18/22', 1),
 ('Bob Dole', '2/17/22', 14),
 ('Bob Dole', '2/15/22', 6),
 ('Bob Dole', '2/14/22', 23),
 ('Bob Dole', '2/11/22', 6),
 ('Bob Dole', '2/10/22', 4),
 ('Bob Dole', '2/1/22', 7),
 ('Neil Patrick Harris', '12/8/21', 16),
 ('Bob Dole', '12/7/21', 1),
 ('Bob Dole', '12/29/21', 1),
 ('Bob Dole', '12/23/21', 7),
 ('Reeves Reeves', '12/13/21', 20),
 ('Reeves Reeves', '11/8/21', 9),
 ('Neil Patrick Harris', '11/5/21', 4),
 ('Reeves Reeves', '11/29/21', 8),
 ('Bob Dole', '11/24/21', 2),
 ('Neil Patrick Harris', '11/22/21', 1),
 ('Bob Dole', '11/2/21', 7),
 ('Bob Dole', '11/19/21', 19),
 ('Reeves Reeves', '11/19/21', 21),
 ('Bob Dole', '11/18/21', 27),
 ('Emma Stone', '11/18/21', 20),
 ('Humankind Admin', '11/18/21', 44),
 ('Bob Dole', '11/16/21', 25),
 ('Bob Dole', '11/11/21', 7),
 ('Neil Patrick Harris', '11/10/21', 4),
 ('Bob Dole', '10/7/21', 8),
 ('Humankind Admin', '10/7/21', 1),
 ('Bob Dole', '10/28/21', 11),
 ('Bob Dole', '10/21/21', 22),
 ('Bob Dole', '10/19/21', 20),
 ('Bob Dole', '10/15/21', 1),
 ('Bob Dole', '10/14/21', 17),
 ('Humankind Admin', '10/1/21', 47),
 ('Bob Dole', '1/7/22', 15),
 ('Bob Dole', '1/6/22', 164),
 ('Bob Dole', '1/5/22', 97),
 ('Neil Patrick Harris', '1/5/22', 15),
 ('Bob Dole', '1/31/22', 27),
 ('Bob Dole', '1/28/22', 52),
 ('Humankind Admin', '1/28/22', 1),
 ('Bob Dole', '1/14/22', 13),
 ('Bob Dole', '1/13/22', 7),
 ('Bob Dole', '1/12/22', 9),
 ('Neil Patrick Harris', '1/12/22', 3),
 ('Bob Dole', '1/11/22', 16),
 ('Bob Dole', '1/10/22', 46)]

In [20]:
%%sql

select e.expert,  claimedatdatetime, count(c.expertid)
from customers_4 c
join experts_1 e on e.expertid = c.expertid
group by e.expert,claimedatdatetime
order by count(c.expertid) desc

 * sqlite:///humankind.sqlite
Done.


[('Bob Dole', '16:59', 19),
 ('Bob Dole', '17:34', 19),
 ('Bob Dole', '18:10', 19),
 ('Bob Dole', '16:40', 18),
 ('Bob Dole', '19:12', 18),
 ('Bob Dole', '15:15', 17),
 ('Bob Dole', '16:42', 16),
 ('Bob Dole', '18:12', 16),
 ('Bob Dole', '19:54', 16),
 ('Bob Dole', '21:07', 16),
 ('Bob Dole', '13:08', 15),
 ('Bob Dole', '16:26', 15),
 ('Bob Dole', '16:33', 15),
 ('Bob Dole', '18:09', 15),
 ('Bob Dole', '18:13', 15),
 ('Bob Dole', '19:22', 15),
 ('Bob Dole', '20:28', 15),
 ('Bob Dole', '20:42', 15),
 ('Bob Dole', '15:40', 14),
 ('Bob Dole', '17:37', 14),
 ('Bob Dole', '18:08', 14),
 ('Bob Dole', '18:22', 14),
 ('Bob Dole', '18:39', 14),
 ('Bob Dole', '19:45', 14),
 ('Bob Dole', '19:49', 14),
 ('Bob Dole', '19:57', 14),
 ('Bob Dole', '14:05', 13),
 ('Bob Dole', '15:43', 13),
 ('Bob Dole', '16:34', 13),
 ('Bob Dole', '16:35', 13),
 ('Bob Dole', '17:07', 13),
 ('Bob Dole', '17:08', 13),
 ('Bob Dole', '18:47', 13),
 ('Bob Dole', '21:09', 13),
 ('Bob Dole', '14:45', 12),
 ('Bob Dole', '16:09', 12),
 ('Bob Dole', '16:36', 12),
 ('Bob Dole', '17:09', 12),
 ('Bob Dole', '17:42', 12),
 ('Bob Dole', '18:00', 12),
 ('Bob Dole', '18:23', 12),
 ('Bob Dole', '19:42', 12),
 ('Bob Dole', '19:47', 12),
 ('Bob Dole', '20:17', 12),
 ('Bob Dole', '21:10', 12),
 ('Bob Dole', '21:30', 12),
 ('Bob Dole', '22:05', 12),
 ('Bob Dole', '16:05', 11),
 ('Bob Dole', '16:56', 11),
 ('Bob Dole', '17:01', 11),
 ('Bob Dole', '17:33', 11),
 ('Bob Dole', '17:39', 11),
 ('Bob Dole', '18:14', 11),
 ('Bob Dole', '18:18', 11),
 ('Bob Dole', '18:31', 11),
 ('Bob Dole', '19:25', 11),
 ('Bob Dole', '20:33', 11),
 ('Bob Dole', '20:46', 11),
 ('Bob Dole', '20:48', 11),
 ('Bob Dole', '15:30', 10),
 ('Bob Dole', '15:44', 10),
 ('Bob Dole', '15:51', 10),
 ('Bob Dole', '16:39', 10),
 ('Bob Dole', '16:43', 10),
 ('Bob Dole', '16:55', 10),
 ('Bob Dole', '17:49', 10),
 ('Bob Dole', '18:32', 10),
 ('Bob Dole', '18:37', 10),
 ('Bob Dole', '18:41', 10),
 ('Bob Dole', '18:52', 10),
 ('Bob Dole', '18:54', 10),
 ('Bob Dole', '18:59', 10),
 ('Bob Dole', '19:15', 10),
 ('Bob Dole', '19:48', 10),
 ('Bob Dole', '21:00', 10),
 ('Bob Dole', '21:37', 10),
 ('Bob Dole', '21:38', 10),
 ('Bob Dole', '13:06', 9),
 ('Bob Dole', '14:25', 9),
 ('Bob Dole', '15:29', 9),
 ('Bob Dole', '16:25', 9),
 ('Bob Dole', '16:37', 9),
 ('Bob Dole', '17:00', 9),
 ('Bob Dole', '17:40', 9),
 ('Bob Dole', '17:46', 9),
 ('Bob Dole', '18:07', 9),
 ('Bob Dole', '18:15', 9),
 ('Bob Dole', '18:21', 9),
 ('Bob Dole', '18:33', 9),
 ('Bob Dole', '18:35', 9),
 ('Bob Dole', '18:40', 9),
 ('Bob Dole', '19:13', 9),
 ('Bob Dole', '20:10', 9),
 ('Bob Dole', '20:12', 9),
 ('Bob Dole', '20:27', 9),
 ('Bob Dole', '20:53', 9),
 ('Bob Dole', '21:08', 9),
 ('Humankind Admin', '19:29', 9),
 ('Bob Dole', '14:00', 8),
 ('Bob Dole', '14:40', 8),
 ('Bob Dole', '14:48', 8),
 ('Bob Dole', '14:49', 8),
 ('Bob Dole', '15:42', 8),
 ('Bob Dole', '16:01', 8),
 ('Bob Dole', '16:32', 8),
 ('Bob Dole', '16:47', 8),
 ('Bob Dole', '17:12', 8),
 ('Bob Dole', '17:25', 8),
 ('Bob Dole', '17:35', 8),
 ('Bob Dole', '18:06', 8),
 ('Bob Dole', '18:19', 8),
 ('Bob Dole', '18:24', 8),
 ('Bob Dole', '19:10', 8),
 ('Bob Dole', '19:18', 8),
 ('Bob Dole', '19:37', 8),
 ('Bob Dole', '19:39', 8),
 ('Bob Dole', '20:05', 8),
 ('Bob Dole', '20:23', 8),
 ('Bob Dole', '20:30', 8),
 ('Bob Dole', '20:54', 8),
 ('Bob Dole', '21:02', 8),
 ('Bob Dole', '22:34', 8),
 ('Humankind Admin', '19:27', 8),
 ('Bob Dole', '14:02', 7),
 ('Bob Dole', '15:04', 7),
 ('Bob Dole', '15:16', 7),
 ('Bob Dole', '15:27', 7),
 ('Bob Dole', '16:00', 7),
 ('Bob Dole', '16:08', 7),
 ('Bob Dole', '16:27', 7),
 ('Bob Dole', '17:11', 7),
 ('Bob Dole', '17:23', 7),
 ('Bob Dole', '17:24', 7),
 ('Bob Dole', '17:53', 7),
 ('Bob Dole', '17:54', 7),
 ('Bob Dole', '17:55', 7),
 ('Bob Dole', '18:20', 7),
 ('Bob Dole', '18:29', 7),
 ('Bob Dole', '18:38', 7),
 ('Bob Dole', '19:04', 7),
 ('Bob Dole', '19:09', 7),
 ('Bob Dole', '19:14', 7),
 ('Bob Dole', '20:47', 7),
 ('Bob Dole', '21:14', 7),
 ('Bob Dole', '21:36'